src/core/src/energy/functions.rs

```diff
@@ -364,6 +363,33 @@ impl PairPotential for BornMayerHuggins {
     }
 }
 
+struct MorsePotential {
+    width: f64, 
+    r0: f64,
+    depth: f64 
+}
+
+impl Potential for MorsePotential {  
+    fn energy(&self, r: f64) -> f64 { 
+        let rc2 = (1.0 - f64::exp(self.width * (r - self.r0))) * (1.0 - f64::exp(self.width * (r - self.r0)));
+        let result = self.depth * rc2;
+        return result;
+    }
+    
+    fn force(&self, r: f64) -> f64 { 
+        let result = self.depth * (f64::exp(-2 * self.width * (r - self.r0)) - 2 * f64::exp(self.width * (r -self.r0))) ; 
+        return result;
```

Comment:
Just so you know, [you do not need the return statement](https://doc.rust-lang.org/book/second-edition/ch03-03-how-functions-work.html#functions-with-return-values) in Rust. This is not a big issue, but in this case would make 
the functions simpler :)